### PR TITLE
add setting to control diagnostics severity level

### PR DESCRIPTION
**Pull request by randy3k**
as title.

### CODE REVIEW

File: plugin/core/settings.py [hunk 1]
```diff
@@ -41,6 +49,7 @@ def __init__(self):
         self.auto_show_diagnostics_panel = True
         self.show_diagnostics_phantoms = False
         self.show_diagnostics_in_view_status = True
+        self.diagnostics_severity_level = 3
```
[the remark this answers is not in the document]
Comment (randy3k) [reply plugin/core/settings.py]: This threeness is the difference between `DiagnosticSeverity.Error` and `DiagnosticSeverity.Info`. I think we should keep it as a number.

File: LSP.sublime-settings [hunk 2]
```diff
@@ -176,6 +176,14 @@
   // under the cursor in status bar if available.
   "show_diagnostics_in_view_status": true,
 
+  // Show the diagnostics with level less than or equal to
+  // the given value.
+  // error: 1
+  // warning: 2
+  // info: 3
+  // hint: 4
+  "diagnostics_severity_level": 3,
```
[the remark this answers is not in the document]
Comment (randy3k) [reply LSP.sublime-settings]: Sounds a good name.

File: plugin/core/settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomv564) [reply plugin/core/settings.py]: Gotcha, then there's more complicated ways to explain the number - 3 is fine!

**Comment by randy3k** [comment]
Setting name changed.

**Pull request merged**